apps/andi/test/unit/andi_web/live/edit_live_view_test.exs

```diff
@@ -0,0 +1,68 @@
+defmodule AndiWeb.EditLiveViewTest do
+  use AndiWeb.ConnCase
+  use Phoenix.ConnTest
+  import Phoenix.LiveViewTest
+
+  alias Andi.DatasetCache
+
+  alias SmartCity.TestDataGenerator, as: TDG
+
+  @url_path "/datasets/"
+
+  setup do
+    GenServer.call(DatasetCache, :reset)
+  end
+
+  describe "Enter Metadata" do
+    test "displays title field", %{conn: conn} do
+      dataset = TDG.create_dataset([])
+      DatasetCache.put(dataset)
+
+      assert {:ok, _view, html} = live(conn, @url_path <> dataset.id)
+      subject = Floki.find(html, ".metadata-form") |> Floki.raw_html()
+
+      assert subject =~ "Title of Dataset"
+      assert subject =~ dataset.business.dataTitle
+    end
+
+    test "display Level of Access as public when private is false", %{conn: conn} do
+      dataset = TDG.create_dataset(%{technical: %{private: false}})
+      DatasetCache.put(dataset)
+
+      assert {:ok, _view, html} = live(conn, @url_path <> dataset.id)
+      [subject] = Floki.find(html, "#metadata_private") |> Floki.attribute("value")
+
+      assert subject =~ "Public"
+    end
+
+    test "display Level of Access as private when private is true", %{conn: conn} do
+      dataset = TDG.create_dataset(%{technical: %{private: true}})
+      DatasetCache.put(dataset)
+
+      assert {:ok, _view, html} = live(conn, @url_path <> dataset.id)
+      [subject] = Floki.find(html, "#metadata_private") |> Floki.attribute("value")
+
+      assert subject =~ "Private"
+    end
+
+    test "adds commas between keywords", %{conn: conn} do
+      dataset = TDG.create_dataset(%{business: %{keywords: ["one", "two", "three"]}})
+      DatasetCache.put(dataset)
+
+      assert {:ok, _view, html} = live(conn, @url_path <> dataset.id)
+      [subject] = Floki.find(html, "#metadata_keywords") |> Floki.attribute("value")
+
+      assert subject =~ "one, two, three"
+    end
+
+    test "should show empty string if keywords is nil", %{conn: conn} do
+      dataset = TDG.create_dataset(%{business: %{keywords: nil}})
+      DatasetCache.put(dataset)
+
+      assert {:ok, _view, html} = live(conn, @url_path <> dataset.id)
+      [subject] = Floki.find(html, "#metadata_keywords") |> Floki.attribute("value")
+
+      assert subject == ""
+    end
+  end
+end
```

Comment:
I thought I added this comment earlier, but looks like it got lost:

Would it be worth adding a Checkov-style test just to verify that all of the other 'pass-through' metadata fields are showing up on the form?  Again - doesn't seem like that much squeeze for a decent amount of juice.

Reply:
We'll do this as well